### PR TITLE
adding sc related files for hardhat testing

### PR DESCRIPTION
**Pull request by architarora**

### CODE REVIEW

File: sc/contracts/1_Storage.sol [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity >=0.7.0 <0.9.0;
+
+/**
+ * @title Storage
+ * @dev Store & retrieve value in a variable
+ * @custom:dev-run-script ./scripts/deploy_with_ethers.ts
+ */
+contract Storage {
```
Comment (midgerate) [inline sc/contracts/1_Storage.sol]: Remove this file

File: sc/contracts/2_Owner.sol [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+pragma solidity >=0.7.0 <0.9.0;
+
+import "hardhat/console.sol";
+
+/**
+ * @title Owner
+ * @dev Set & change owner
+ */
+contract Owner {
```
Comment (midgerate) [inline sc/contracts/2_Owner.sol]: Remove this file

File: sc/contracts/3_Ballot.sol [hunk 3]
```diff
@@ -0,0 +1,138 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity >=0.7.0 <0.9.0;
+
+/** 
+ * @title Ballot
+ * @dev Implements voting process along with vote delegation
+ */
+contract Ballot {
```
Comment (midgerate) [inline sc/contracts/3_Ballot.sol]: Remove this file

File: sc/contracts/mintcontract.sol [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    /**
+     * @notice See {ERC721-baseURI}.
+     */
+    function _baseURI() internal view virtual override returns (string memory) {
+        return "ipfs://";
+    }
+
+    function mint(uint256 _tokenId, string memory _tokenURI) external payable {
```
Comment (midgerate) [inline sc/contracts/mintcontract.sol]: This wont work. The client will just pass the number of tokens to mint (1,2,3) and contract should do the rest. 

Currently you are relying on the client to send the tokenIds which can be manipulated easily.

File: sc/contracts/1_Storage.sol (hunk 1 above)
Comment (architarora) [reply sc/contracts/1_Storage.sol]: Removed

File: sc/contracts/2_Owner.sol (hunk 2 above)
Comment (architarora) [reply sc/contracts/2_Owner.sol]: Removed

File: sc/contracts/3_Ballot.sol (hunk 3 above)
Comment (architarora) [reply sc/contracts/3_Ballot.sol]: Removed

File: sc/contracts/mintcontract.sol (hunk 4 above)
Comment (architarora) [reply sc/contracts/mintcontract.sol]: Changed to amount param instead

**Pull request merged**